### PR TITLE
Reject if request for close down is invalid

### PR DESCRIPTION
**Pull request by CBrohus**
Accept close down request
Reject close down if date is invalid
Reject close down if gsrn is missing
Reject close down if transaction id is missing
Reject if metering point does not exist

## References

* #565

### CODE REVIEW

**Review by kft — DISMISSED** [review]

File: source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.MeteringPoints.Application.Common;
+
+namespace Energinet.DataHub.MeteringPoints.Application.CloseDown
+{
+    public record RequestCloseDown(
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs:19]: Nit: Could adher to other similar requests naming (CreateMeteringPointRequest, DisconnectMeteringPointRequest etc)

File: source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        string TransactionId,
+        string GsrnNumber,
+        string EffectiveDate) : IBusinessRequest;
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs:22]: We have an interface called IChangeMeteringPointRequest which are used for every other request which changes a metering point

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs [hunk 3]
```diff
@@ -34,6 +34,8 @@ public sealed class DocumentType : EnumerationType
         public static readonly DocumentType RejectDisconnectMeteringPoint = new(10, nameof(RejectDisconnectMeteringPoint), typeof(RejectMessage));
         public static readonly DocumentType ConfirmReconnectMeteringPoint = new(11, nameof(ConfirmReconnectMeteringPoint), typeof(ConfirmMessage));
         public static readonly DocumentType RejectReconnectMeteringPoint = new(12, nameof(RejectReconnectMeteringPoint), typeof(RejectMessage));
+        public static readonly DocumentType AcceptCloseDownRequest = new(13, nameof(AcceptCloseDownRequest), typeof(ConfirmMessage));
+        public static readonly DocumentType RejectCloseDownRequest = new(14, nameof(RejectCloseDownRequest), typeof(RejectMessage));
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs:38]: ConfirmCloseDownMeteringPoint & RejectCloseDownMeteringPoint?

**Review by Korgath — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs (hunk 1 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs:19]: This is also reflected down in the naming of handlers etc

File: source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs (hunk 2 above)
Comment (kft) [reply source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs:22]: Yes. Currently not needed here.

File: source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs (hunk 1 above)
Comment (kft) [reply source/Energinet.DataHub.MeteringPoints.Application/CloseDown/RequestCloseDown.cs:19]: It does. It is a "Request" for close down. Not the actual close down action itself, which will be added later on.

File: source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs (hunk 3 above)
Comment (kft) [reply source/Energinet.DataHub.MeteringPoints.Infrastructure/EDI/DocumentType.cs:38]: Actually not. "Accept" and "Reject" is language of the business. I have never heard them use the word "Confirm" in this regard.

**Pull request merged**